spit/classifier.py

```diff
@@ -53,12 +54,73 @@ def init_session(self):
     self.y_train = keras.utils.to_categorical(list(self.label_dict.values()), 
                                               num_classes=len(self.label_dict))
     # add optimizer, learning rate, and loss function
-    self.model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'], rate=1e-4)
+    adam = optimizers.Adam(lr=0.001, beta_1=0.9, beta_2=0.999, epsilon=None, decay=0.0, amsgrad=False)
+
+    self.model.compile(loss='categorical_crossentropy', optimizer=adam, metrics=['accuracy'])#, rate=1e-4)
     return
   
+    
+  # This method might need to be modified if the comparison of accuracies
+  # is done while training
+  def save_model(self, file_name=None, test_dset=None, test_labels=None, comparing=False):
+    '''
+    Method save_model():
+    To save a model. Either as a model itself
 
-# Other architectures
-    """
+    Parameters:
+
+    file_name: The name of the model you want to save as can overwrite previous one
+                should be a string in the format of 'blah.h5'
+    test_dset/test_labels: test data in order to compare the accuracies
+    comparing: either it is to save as a checkpoint of the original model
+              or to compare with a previous model and save the better one
+
+    '''
+    saving_dir = '/data/checkpoints/' # may need to change, depends on where to run the scripts
```

Comment:
Let's make this an optional parameter and have it default to the local directory, i.e   './'